### ISSUE TITLE
[ffmpeg error on RPi2 Raspbian Jessie] -bash: -threads: command not found

### ISSUE THREAD

**Issue by Luc-S** [KhaosT/homebridge-camera-ffmpeg#40]
I'm running into an issue when adding a NEO Coolcam to HomeKit using homebridge on my RPi2. The accessory is added but I only see a the grey videorecorder icon indicating there's no image available.

The IP camera URL is working in VLC (on a Mac) so I ran the command below in the RPi2 terminal (this is the command including parameters thrown in the console by homebridge when trying to open the camera accessory in HomeKit):

`
pi@homesystem:~ $ ffmpeg -re -i http://192.168.1.251:8001/videostream.cgi?user=xx&pwd=xx -threads 0 -vcodec libx264 -an -pix_fmt yuv420p -r 25 -f rawvideo -tune zerolatency -vf scale=640:360 -b:v 132k -bufsize 132k -payload_type 99 -ssrc 1 -f rtp -srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params WKD4mXCDsn4/5lYxUq250YSg4Yq9jdhwPxkwIqOP srtp://192.168.1.112:64874?rtcpport=64874&localrtcpport=64874&pkt_size=1378
`

This outputs:

[2] 1396
[3] 1397
-bash: -threads: command not found
[4] 1398
[3]   Exit 127                pwd=xx -threads 0 -vcodec libx264 -an -pix_fmt yuv420p -r 25 -f rawvideo -tune zerolatency -vf scale=640:360 -b:v 132k -bufsize 132k -payload_type 99 -ssrc 1 -f rtp -srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params WKD4mXCDsn4/5lYxUq250YSg4Yq9jdhwPxkwIqOP srtp://192.168.1.112:64874?rtcpport=64874
[4]-  Done                    localrtcpport=64874

Any thoughts are much appreciated :) thanks!

**Comment by NorthernMan54** [comment]
Looks like a copy and paste error.   When you pasted it into raspbian it went across multiple lines, when it is a single line.

**Comment by Luc-S** [comment]
It is a single line however I do think you're making a valid point by saying ffmpeg doesn't execute is as such. 

[3] Exit 127 pwd=xx

It's like the & character isn't recognised OK.

**Comment by gijoecool** [comment]
Try replacing the '&'s with '%26'.

**Comment by Luc-S** [comment]
That brings me a step further in the process but I'm not quite there yet. When running the command:
`
pi@homesystem:~/plugins $ ffmpeg -re -i http://192.168.1.251:8001/videostream.cgi?user=xx%26pwd=xx -threads 0 -vcodec libx264 -an -pix_fmt yuv420p -r 25 -f rawvideo -tune zerolatency -vf scale=640:360 -b:v 132k -bufsize 132k -payload_type 99 -ssrc 1 -f rtp -srtp_out_suite AES_CM_128_HMAC_SHA1_80 -srtp_out_params /BmWKBnHi8V/tKqqLDQl5ftz2ofNurBQo3OO74Gz srtp://192.168.1.112:54518?rtcpport=54518&localrtcpport=54518&pkt_size=1378
`

I now get the result below.

[6] 2246
[7] 2247
pi@homesystem:~/plugins $ ffmpeg version 2.7.2 Copyright (c) 2000-2015 the FFmpeg developers
  built with gcc 4.9.2 (Raspbian 4.9.2-10)
  configuration: --enable-gpl --enable-postproc --enable-swscale --enable-avfilter --enable-libmp3lame --enable-libvorbis --enable-libtheora --enable-libx264 --enable-libspeex --enable-shared --enable-pthreads --enable-libopenjpeg --enable-libfaac --enable-nonfree
  WARNING: library configuration mismatch
  avutil      configuration: --arch=arm .... 

Followed by what appears to be all parameters.

**Comment by gijoecool** [comment]
You still have two ampersands in the srtp URL...

**Comment by Luc-S** [comment]
Ah totally missed those! Now we have some info we can work with, it now gives me:

[http @ 0xcd16e0] HTTP error 401 Unauthorized
http://192.168.1.251:8001/videostream.cgi?user=xx%26pwd=xx: Input/output error

**Comment by gijoecool** [comment]
Can you access that URL from that machine?

**Comment by Luc-S** [comment]
On my Mac in VLC it now asks credentials (it doesn't when using the & in that url between user and pwd parameters).

On my Mac in Safari it works fine without asking me for credentials.

On the RPi2 in the default browser in Raspbian it asks me for credentials again.

**Comment by gijoecool** [comment]
Can you use the format http://user:password@192.168.1.251:8001/videostream.cgi ?

**Comment by Luc-S** [comment]
Sadly no. But I see what you are after, I'll try to search on the web and through the ipcams' source code to see if I have any alternative URL's to try.

**Comment by Luc-S** [comment]
Ok I think I'm very close now. I have installed libav-tools and created a link between ffmpeg and avconv as per issue #21. Now it looks like the stream is working when running the ffmpeg command in terminal.

In homebridge it's still not working but this happened; I connected to the camera in HomeKit and it got stuck on loading without image, homebridge gave me the ffmpeg command in console which I started in a new terminal window manually. Guess what! The stream started on my phone in HomeKit!

So I guess I now have to find out the difference between manually starting the stream in terminal and how the plugin starts it.
